src/Language/PureScript/Interactive/Parser.hs

```diff
@@ -112,3 +114,12 @@ parseReplQuery' str =
     Nothing -> Left ("Don't know how to show " ++ str ++ ". Try one of: " ++
                       intercalate ", " replQueryStrings ++ ".")
     Just query -> Right query
+
+-- | To show error message when 'let' is used for declaration in PSCI,
+-- which is deprecated.
+psciDeprecatedLet :: P.TokenParser Command
+psciDeprecatedLet = do
+  P.reserved "let"
+  P.indented
+  _ <- mark (many1 (same *> P.parseLocalDeclaration))
```

Comment:
You could just use `_ <- parseLocalDeclaration` here. One declaration is enough to fail.